app/Http/Controllers/RestApi/RestApiMember.php

```diff
@@ -98,10 +153,234 @@ public function getChanged( $request, $revisionId, $is_admin = false ) {
+			} else {
+				$data[ $id ] = ApiHelper::getMemberAsArray( $member, $allowedGroups, $is_admin );
+			}
 		}
 
 		return json_encode( $data );
 	}
+
+	/**
+	 * Update the given member.
+	 *
+	 * @param Request $request - the http Request with the member data
+	 * @param number $memberId
+	 *
+	 * @return int member id
+	 *
+	 * @throws IllegalFieldUpdateMode
+	 * @throws \App\Exceptions\GroupNotFoundException
+	 * @throws \App\Exceptions\IllegalArgumentException
+	 * @throws \App\Exceptions\InvalidFixedValueException
+	 * @throws \App\Exceptions\MemberNotFoundException
+	 * @throws \App\Exceptions\MemberUnknownFieldException
+	 * @throws \App\Exceptions\MultiSelectOverwriteException
+	 * @throws \App\Exceptions\NoGroupException
+	 * @throws \App\Exceptions\ValueTypeException
+	 * @throws \App\Exceptions\WeblingAPIException
+	 * @throws \App\Exceptions\WeblingFieldMappingConfigException
+	 * @throws \Webling\API\ClientException
+	 * @throws BadRequestException
+	 */
+	public function updateMember( Request $request, $memberId ) {
+		ApiHelper::checkIntegerInput( $memberId );
+		$memberRepo = ApiHelper::createMemberRepo( $request->header( $key = 'db_key' ) );
+
+		$member = $memberRepo->get( $memberId );
+		ApiHelper::assertAllowedMember( ApiHelper::getAllowedGroups( $request ), $member );
+
+		$memberData = $this->extractMemberData( $request );
+		$patched    = $this->patchMember( $request, $member, $memberData );
+
+		return $memberRepo->save( $patched )->id;
+	}
+
+	/**
+	 * Update or insert the given member.
+	 *
+	 * @param Request $request - the http Request with the member data
+	 *
+	 * @return int member id
+	 *
+	 * @throws BadRequestException
+	 * @throws IllegalFieldUpdateMode
+	 * @throws \App\Exceptions\GroupNotFoundException
+	 * @throws \App\Exceptions\IllegalArgumentException
+	 * @throws \App\Exceptions\InvalidFixedValueException
+	 * @throws \App\Exceptions\MemberNotFoundException
+	 * @throws \App\Exceptions\MemberUnknownFieldException
+	 * @throws \App\Exceptions\MultiSelectOverwriteException
+	 * @throws \App\Exceptions\NoGroupException
+	 * @throws \App\Exceptions\ValueTypeException
+	 * @throws \App\Exceptions\WeblingAPIException
+	 * @throws \App\Exceptions\WeblingFieldMappingConfigException
+	 * @throws \Webling\API\ClientException
+	 */
+	public function upsertMember( Request $request ) {
+		$memberData = $this->extractMemberData( $request );
+
+		// update the given member, if it contains an id
+		if ( ! empty( $memberData[ Member::KEY_ID ] ) ) {
+			$idField  = $memberData[ Member::KEY_ID ];
+			$memberId = isset( $idField['value'] ) ? $idField['value'] : $idField;
+
+			return $this->updateMember( $request, $memberId );
+		}
+
+		$member  = new Member();
+		$patched = $this->patchMember( $request, $member, $memberData, true );
+
+		$memberRepo = ApiHelper::createMemberRepo( $request->header( $key = 'db_key' ) );
+		$match      = $memberRepo->findExisting( $member, ApiHelper::getAllowedGroups( $request ) );
```

Comment:
Kann es sein, dass hier nur diejenigen Member gefunden werden, auf die der User Zugang hat?
Dann kann es aber Situationen geben, in denen ein Member existiert, dieser aber nicht gefunden werden kann, weil der derzeit 'Anfragende' nicht die Berechtigung hat, den bestehenden Member zu sehen.

Reply:
Ja, das ist genau so. Da ein Member gleichzeitig in einem Kanton Mitglied sein kann, in einem anderen aber nur Sympi und wir dies nicht anders als mit zwei Datensätzen abbilden können, ist dieses Verhalten gewünscht. Natürlich ist es höchst suboptimal, aber mit Webling sehe ich keine andere Möglichkeit.